pkg/devspace/config/loader/validate.go

```diff
@@ -212,6 +212,9 @@ func validateHooks(config *latest.Config) error {
 		if hookConfig.Wait != nil && !hookConfig.Wait.Running && hookConfig.Wait.TerminatedWithCode == nil {
 			return errors.Errorf("hooks[%d].wait.running or hooks[%d].wait.terminatedWithCode is required if hooks[%d].wait is used", index, index, index)
 		}
+		if hookConfig.Container.ContainerName != "" && hookConfig.Container.ImageSelector == "" && len(hookConfig.Container.LabelSelector) == 0 {
```

Comment:
These need to be separte checks:
1. containerName != "" && len(labelSelector) == 0
2. len(labelSelector) == 0 && imageSelector == ""